src/plugins/intel_npu/tests/unit/npuw/serialization.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include "intel_npu/config/config.hpp"
+#include "intel_npu/config/npuw.hpp"
+#include "openvino/openvino.hpp"
+#include "serialization.hpp"
+#include "compiled_model.hpp"
+#include "spatial.hpp"
+#include "model_generator/model_generator.hpp"
+
+TEST(SerializationTest, BasicTypes_string) {
+    using namespace ov::npuw::s11n;
+
+    std::string var("NPUW");
+    std::string res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_bool) {
+    using namespace ov::npuw::s11n;
+
+    bool var = true;
+    bool res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_float) {
+    using namespace ov::npuw::s11n;
+
+    float var = 3.1415;
+    float res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, BasicTypes_streampos) {
+    using namespace ov::npuw::s11n;
+
+    std::stringstream buf;
+    buf.write("NPUW", 4);
+
+    std::streampos var = buf.tellp();
+    std::streampos res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(var, res);
+}
+
+TEST(SerializationTest, OVTypes_Tensor) {
+    using namespace ov::npuw::s11n;
+
+    std::vector<uint8_t> data {0, 1, 2, 3};
+    ov::Tensor var(ov::element::u8, ov::Shape({2, 2}), data.data());
+    ov::Tensor res;
+
+    std::stringstream ss;
+
+    write(ss, var);
+    read(ss, res);
+
+    EXPECT_EQ(res.get_element_type(), ov::element::u8);
+    EXPECT_EQ(res.get_shape(), ov::Shape({2, 2}));
+
+    std::vector<uint8_t> data_res(4, 0);
+    std::memcpy(data_res.data(), res.data(), 4);
+
+    EXPECT_EQ(data, data_res);
```

Comment:
Do we have more common function to compare `ov::Tensor` content in OpenVINO?

Reply:
I couldn't find anything